nvtabular/io.py

```diff
@@ -617,30 +617,18 @@ def _write_metadata(meta_list):
 @annotate("write_output_partition", color="green", domain="nvt_python")
 def _write_output_partition(gdf, processed_path, shuffle, out_files_per_proc, fs):
     gdf_size = len(gdf)
-    if shuffle == "full":
-        # Dont need a real sort if we are doing in memory later
-        typ = np.min_scalar_type(out_files_per_proc * 2)
-        ind = cp.random.choice(cp.arange(out_files_per_proc, dtype=typ), gdf_size)
-        result = group_split_dispatch(gdf, ind, out_files_per_proc, ignore_index=True)
-        del ind
-        del gdf
-        # Write each split to a separate file
-        for s, df in result.items():
-            prefix = fs.sep.join([processed_path, "split." + str(s)])
-            pw = get_cache().get_pq_writer(prefix, s, mem=True)
-            pw.write_table(df)
-    else:
+    out_files_per_proc = out_files_per_proc or 1
+    if shuffle and shuffle != "full":
```

Comment:
I think if shuffle == 'full' is passed in we should throw a ```NotImplementedError``` rather than just not shuffle the data at all.

Reply:
`shuffle == "full"` is currently a full shuffle per worker - So, it is implemented.  The reason we don't sort here, is becuase we are going to read back the in-memory file and sort again before persisting to disk.